src/Utility/Inflector.php

```diff
@@ -274,10 +274,18 @@ public static function pluralize(string $word): string
             static::$_cache['irregular']['pluralize'] = '(?:' . implode('|', array_keys(static::$_irregular)) . ')';
         }
 
-        if (preg_match('/(.*?(?:\\b|_))(' . static::$_cache['irregular']['pluralize'] . ')$/i', $word, $regs)) {
-            static::$_cache['pluralize'][$word] = $regs[1] . substr($regs[2], 0, 1) .
+        $inflection = static::_detectInflection($word);
+        $normalizedWord = $word;
+        if ($inflection) {
+            $normalizedWord = Inflector::underscore($word);
+        }
+
+        if (preg_match('/(.*?(?:\\b|_))(' . static::$_cache['irregular']['pluralize'] . ')$/i', $normalizedWord, $regs)) {
```

Comment:
Could this pattern `(?:\\b|_)` not be modified to `(?:\\b|_|[A-Z])` instead of incurring additional inflection calls?

Reply:
I checked, 3 failing tests indicate that this might not be enough.